litmus-portal/graphql-server/pkg/cluster/handler/handler.go

```diff
@@ -274,3 +274,20 @@ func SendRequestToSubscriber(subscriberRequest clusterOps.SubscriberRequests, r
 
 	r.Mutex.Unlock()
 }
+
+// GetAgentDetails fetches agent details from the DB
+func GetAgentDetails(ctx context.Context, agentName string, projectID string) (*model.Cluster, error) {
+	cluster, err := dbOperationsCluster.GetAgentDetails(ctx, agentName, projectID)
+	if err != nil {
+		return nil, err
+	}
+
+	newCluster := model.Cluster{}
+
+	err = copier.Copy(&newCluster, &cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	return &newCluster, nil
+}
```

Comment:
can there be multiple agents with the same name ? why cluster id is not being used instead of agent name ?

Reply:
> can there be multiple agents with the same name ? why cluster id is not being used instead of agent name ?  

Agent name will be unique in a project. Agent name and project ID is used for better user experience. When user tries to upgrade the agent, user will have to enter agent name and project ID